### PR TITLE
Feature: Basic dashboard

### PR DESCRIPTION
**Pull request by oxifinch**
- Added basic dashboard 
    - See total amount allocated for this budget period 
    - See total amount remaining for this budget period 
    - See all user-defined categories with their allocated and remaining amount 
- Added models for Categories and BudgetCategories, and implemented relations with existing models

### CODE REVIEW

File: categories/repo.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+func (c *CategoryRepo) Get(id int) (*database.Category, error) {
+	category := database.Category{}
+	err := c.db.First(&category, id).Error
+
+	return &category, err
+}
+
+func (c *CategoryRepo) GetAllWithOwnerId(id int) ([]*database.Category, error) {
```
Comment (fholmqvist) [inline categories/repo.go]: `GetAllWithUserID`

1. Abbreviations are all caps.
2. We have a `User`, not such thing as `Owner`.

File: database/database.go [hunk 2]
```diff
@@ -20,10 +20,67 @@ func NewDatabase() *Database {
...
-	err = db.AutoMigrate(&User{}, &Budget{})
+	err = db.AutoMigrate(&User{}, &Budget{}, &Category{}, &BudgetCategory{})
 	if err != nil {
 		log.Fatalf("NewDatabase: %v\n", err)
 	}
 
 	return &Database{db}
 }
+
+func (db *Database) Seed() {
```
Comment (fholmqvist) [inline database/database.go:30]:
```
log.Warning("\n\nWARNING: Seeding database, dropping all data!\n\n")
```

File: main.go [hunk 3]
```diff
@@ -8,8 +8,8 @@ const port = ":8080"
 
 func main() {
 	db := database.NewDatabase()
+	db.Seed()
```
Comment (fholmqvist) [inline main.go]:
```
// TODO: Args.
const (
    debug = true
)

...

if debug {
    db.Seed()
}
```

File: users/repo.go [hunk 4]
```diff
@@ -13,7 +13,7 @@ func NewUserRepo(db *database.Database) *UserRepo {
 }
 
 func (u *UserRepo) Get(id int) (*database.User, error) {
-	var user database.User
+	user := database.User{}
```
Comment (fholmqvist) [inline users/repo.go]: Change back.

Idiom:
* Undefined = `var`
* Defined = `:=`

File: users/repo.go [hunk 5]
```diff
@@ -27,7 +27,7 @@ func (u *UserRepo) GetAll() (*[]database.User, error) {
 }
 
 func (u *UserRepo) GetByCredentials(username string, password string) (*database.User, error) {
-	var user database.User
+	user := database.User{}
```
Comment (fholmqvist) [inline users/repo.go]: Same as above.

**Review by fholmqvist — COMMENTED** [review]
Quality of life things, minor polish.

File: database/database.go (hunk 2 above)
Comment (oxifinch) [reply database/database.go:30]: log.Warning doesn't exist, using printf instead.

**Review by fholmqvist — APPROVED** [review]

**Pull request merged**